Quarto docker (#2625)
tenzir/vast: docker/quarto/README.md

# Quarto runner

> In this directory we define a dockerized environment to run Quarto.

Reviewer: What's the features we get on top of the official Quarto Docker image?

Author: fair point, let me complete this

# Quarto runner

In this directory we define a dockerized environment to run Quarto. The image
builds on top of the [VAST image](../../Dockerfile) and adds R, Poetry, and
Quarto. This creates a runtime suitable for calling the VAST binary from
notebooks.

We don't publish the Quarto Docker image, so you need to build it locally.

The `docker-compose.bind.yaml` overlay contains the configurations to bind mount
the VAST repository. The provided commands are executed as the `vast:vast` user
but the resulting output files will be created from the user defined in the

> `HOST_UID` and `HOST_GID` environment variables. From this directory:

Author: same as https://github.com/tenzir/vast/pull/2625#discussion_r1002976863, does it help?

> 

Author: I simplified this and added a `to do X, run this` formulation:
```markdown
The `docker-compose.bind.yaml` overlay contains the configurations to bind mount
the VAST repository on the Quarto container and avoid file system permission
conflicts between the container and the host users. For instance, to run the
Quarto build in the `web` directory, open a terminal in the directory containing
this README and run the following:
```

# Quarto runner

In this directory we define a dockerized environment to run Quarto. The image

> build on top of the VAST image, adding an R installation, Poetry, and of course
> Quarto. This makes it possible to call the VAST binary from within the
> notebooks.

Reviewer: ```suggestion
builds on top of the VAST image and adds R, Poetry, and Quarto. This creates a
runtime suitable for calling the VAST binary from notebooks.
```

Can we link the VAST image?

Author: the repository or the dockerfile?

Reviewer: Whatever makes most sense. As a reader of this text, I was longing for an option to click when I read "VAST image", along the lines of "please take me to it".

Author: done

> We don't publish the Quarto Docker image, so users are required to build it
> locally.

Reviewer: ```suggestion
We don't publish the Quarto Docker image, so you need to build it locally.
```

No shyness, talk to your audience directly!

# Quarto runner

In this directory we define a dockerized environment to run Quarto. The image
builds on top of the [VAST image](../../Dockerfile) and adds R, Poetry, and
Quarto. This creates a runtime suitable for calling the VAST binary from
notebooks.

We don't publish the Quarto Docker image, so you need to build it locally.

> The `docker-compose.bind.yaml` overlay contains the configurations to bind mount

Reviewer 2: Please also put all newly added docker-compose files into the list in `docker/README.md`.

Author: Oops, missed this. While filling it, I also realized that the default `quarto/docker-compose.yaml` didn't make much sense at it, so I added back the default layer in case you don't use the bind mount.

> The `docker-compose.bind.yaml` overlay contains the configurations to bind mount

Reviewer: ```suggestion
The `docker-compose.bind.yaml` overlay contains the configurations to bind-mount
```

Otherwise this reads like a typo.

> the VAST repository on the Quarto container and avoid file system permission

Reviewer: ```suggestion
the VAST repository on the Quarto container and avoid filesystem permission
```
